subprojects/plugins/src/main/java/org/gradle/api/plugins/jvm/internal/DefaultJvmTestSuite.java

```diff
@@ -302,6 +306,16 @@ public void useTestNG(Provider<String> version) {
     }
 
     private void setFrameworkTo(Frameworks framework, Provider<String> versionProvider) {
+        this.targets.withType(JvmTestSuiteTarget.class).configureEach(target -> {
+            target.getTestTask().configure(task -> {
+                if (task.getOptionsAccessed()) {
+                    throw new GradleException(String.format("You cannot set the test framework on suite: %s after accessing test options on an associated Test task: %s.",
```

Comment:
The error message in the `Test` task prints out the type of the framework we are trying to set. Do you think thats worth printing here as well?

Reply:
My first thought was that existing Test tasks are everywhere, and are configured in many different ways, so seeing more detail when the error occurs on that side would help track down misconfigurations.  Test Suites are newer, and hopefully haven't seen a lot of complex usage where it would be hard to track down the source of the change when the error occurs on this side.

But thinking about it a little more, I don't think that's a good objection.  I'd want to see the name.  I'll update `Frameworks` to have a display name like `TestFramework`.